src/game_api/state.hpp

```diff
@@ -51,25 +51,25 @@ struct StateMemory
     float fadevalue; // 0.0 = all visible; 1.0 = all black
     uint32_t fadeout;
     uint32_t fadein;
-    uint32_t loading_black_screen_timer; // if state.loading is 1, this timer counts down to 0 while the screen is black (used after Ouroboros, in
-                                         // credits, ...)
-    uint8_t ingame;
-    uint8_t playing;
+    uint32_t loading_black_screen_timer; // if state.loading is 1, this timer counts down to 0 while the screen is black (used after Ouroboros, in credits etc.)
+    /// True when you have control over your character
+    bool ingame;

```

Comment:
This is a breaking change unfortunately... Probably not widely used but it is used even in our examples.

Reply:
i fixed it, but the autodoc ignores my `NoDoc` comment :/

Reply:
Right, I didn't add nodoc to the usertype parser cause it was such a hack already to get it working without rewriting the whole thing

Reply:
If you want, you can also think of a better name for these new variables. They sound and behave almost exactly the same but I don't remember the difference.

Reply:
Or at least document it. I think ingame is everything after the menu, and playing is in an actual level.

Reply:
I added NoDoc support, merging soon

Reply:
`playing` actually seems like basically `state.screen == SCREEN.LEVEL`, having control of a character is misleading and you got them wrong way around. How about `in_game` and `in_level`

Reply:
I added the comments, but i see some bugs with them now, like when going to options menu both are set to false, but going back only sets the ingame to true, and the playing stays false until next level

ingame is weird 
After the intro cutscene is set to true until you press the button to go to menu
When going to online lobby, is set to true, but not to false when going back, to set it to false again you can go to character select or Player profile and back (is set to false on the way back, not opening it)
Going to the seed input sets ingame to true, going back (correctly?) sets it to false
Is set to true in arena team select, after selecting team is set to false, then in arena itself to true again, the `playing` stays false

When going first time to the camp, only ingame is set to true, but when you die in a level and go to camp, both are true, even thou playing is set to false on the death screen

Reply:
Some of that weirdness almost seems like a bug in the game, or what even is the function of these variables

Reply:
dunno, a added a break point with x64dbg and they are just written to a static memory, and the only access/write to it seams to be that code. Nothing else

Reply:
If it can't be trusted after going to options etc. seems kinda pointless to have at all and state.screen can be used for the same purpose. Maybe that's what the devs did too and forgot about this...